Testscripts/Linux/TestRDMA_MultiVM.sh

```diff
@@ -488,23 +488,27 @@ function Main() {
 	# It's safer to obtain pkeys in the test script because some distros
 	# may require a reboot after the IB setup is completed
 	# Find the correct partition key for IB communicating with other VM
-	if [ -z ${MPI_IB_PKEY+x} ]; then
-		firstkey=$(cat /sys/class/infiniband/mlx5_0/ports/1/pkeys/0)
-		LogMsg "Getting the first key $firstkey"
-		secondkey=$(cat /sys/class/infiniband/mlx5_0/ports/1/pkeys/1)
-		LogMsg "Getting the second key $secondkey"
-
-		# Assign the bigger number to MPI_IB_PKEY
-		if [ $((firstkey - secondkey)) -gt 0 ]; then
-			export MPI_IB_PKEY=$firstkey
-			echo "MPI_IB_PKEY=$firstkey" >> /root/constants.sh
+	if [[ $is_nd == "no" ]]; then
+		if [ -z ${MPI_IB_PKEY+x} ]; then
+			firstkey=$(cat /sys/class/infiniband/mlx5_0/ports/1/pkeys/0)
```

Comment:
PKEY will always be at index 0.

Reply:
Please add a comment as to what the expected PKEYs are. For example in CX3-Pro the second PKEY has to be 0x7ffff
$pkeys = executeRemoteCommand $ipaddress "cat /sys/class/infiniband/mlx4_0/ports/1/pkeys/*"
    if (!($pkeys[0] -match "0x8[0-9a-fA-F][0-9a-fA-F][0-9a-fA-F]" -and $pkeys[1] -eq "0x7fff"))
    {
        Write-LogError "Unexpected PKEYs in ${$ipaddress}: $pkeys"
        exit 1
    }

Reply:
OK, @friver-ms I will add this verification in another PR. I don't have the bandwidth for this. I will open the issue for future update.

Reply:
#1041